client/attest.go

```diff
@@ -183,6 +191,49 @@ func (d *SevSnpDevice) Close() error {
 	return nil
 }
 
+// CreateTdxDevice opens the TDX attestation driver and wraps it with behavior
+// that allows it to add an attestation quote to pb.Attestation.
+func CreateTdxDevice() (*TdxDevice, error) {
+	d, err := tg.OpenDevice()
+	if err != nil {
+		return nil, err
+	}
+	return &TdxDevice{Device: d}, nil
+}
+
+// AddAttestation will get the TDX attestation quote given opts.TEENonce
+// and add them to `attestation`. If opts.TEENonce is empty, then uses
+// contents of opts.Nonce.
+func (d *TdxDevice) AddAttestation(attestation *pb.Attestation, opts AttestOpts) error {
+	var tdxNonce [tabi.TdReportDataSize]byte
+	if len(opts.TEENonce) == 0 {
+		copy(tdxNonce[:], opts.Nonce[:])
```

Comment:
opts.Nonce is already a slice, so I think [:] is not needed

Also do we care about the size of opts.Nonce (what if it's zero)?

Reply:
Nonce can not be zero. Check for this is already there 
```
if len(opts.Nonce) == 0 {
	return nil, fmt.Errorf("provided nonce must not be empty")
}
```
nonce can be of any size. For teenonce the size is fixed.

Reply:
Yes, slicing is not required. I have changed it.

Reply:
Okay, looks a little bit weird to me because the tdxNonce can be only the first part of the opt.Nonce (may result not enough entropy for the nonce?). I'm not sure maybe @deeglaze and @alexmwu can confirm if it's okay.

Reply:
I have tested the implementation and it is working fine.

And for `copy(tdxNonce[:], opts.Nonce[:])` , if length of nonce is less that 64 then it will concatenate zeros at the end.

Reply:
If the user does not provide a teenonce, then the nonce will be used as the teenonce, either concatenated with zeroes or truncated to make it 64 bytes long.

But If teenonce is present, then it must be 64 bytes in length.